pyctcdecode/language_model.py

```diff
@@ -379,6 +387,66 @@ def load_from_dir(cls, filepath: str) -> "LanguageModel":
         kenlm_model = kenlm.Model(filenames["kenlm"])
         return cls(kenlm_model, unigrams, **json_attrs)
 
+    @classmethod
+    def load_from_hf_hub(cls, pretrained_path: str, kenlm_filename: str = "kenLM.arpa", cache_dir: Optional[str] = None, **kwargs):
+        """Class method to load model from https://huggingface.co/
+
+        Args:
+            pretrained_path: string, the `model id` of a pretrained model hosted inside a model
+                repo on https://huggingface.co/. Valid model ids can be namespaced under a user or
+                organization name, like ``kensho/5gram-spanish-kenLM``. For more information, please
+                take a look at https://huggingface.co/docs/hub/main .
+            kenlm_filename: file name of the kenlm model as expected to be saved on https://huggingface.co/. Defaults to "kenlm.arpa".
+            cache_dir: path to where the language model should be downloaded and cached.
+        """
+        from . import __version__ as VERSION, __package_name__ as LIBRARY_NAME
+
+        CACHE_DIRECTORY = cache_dir or os.path.join(Path.home(), ".cache", LIBRARY_NAME)
+
+        try:
+            from huggingface_hub import hf_hub_download
+        except ImportError:
+            raise ImportError(
+                "You need to install huggingface_hub to use `load_from_hf_hub`. "
+                "See https://pypi.org/project/huggingface-hub/ for installation."
+            )
+
+        # download and cache kenLM model
+        kenlm_model_path = hf_hub_download(
+            repo_id=pretrained_path,
+            filename=kenlm_filename,
```

Comment:
im a bit confused here- why is the kenlm path taken as an argument and the unigrams and attributes filenames are take as constants from the class variable?

Reply:
The idea was that users might want to upload both a `.bin` and a `.ar` file. But it might indeed be better to just take a hard-coded name for the Ken lm language model instead. Maybe let's just call the file `lm_model.ken` no matter whether it's binary or `.ar`? 

Or what would you prefer here?

Reply:
I thought a bit about it and I think, a clean solution is to provide a `model_format` function argument that is either `.arpa` or `.bin`. I've also added an `HTTPError` check

Reply:
Let me know what you think